torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -201,10 +200,39 @@ class ShapePropagator {
         continue;
       }
     }
-
     return tensor_types;
   }
 
+  c10::optional<c10::ScalarType> typeToScalarType(const c10::TypePtr & type) {
+    if(auto cast = type->cast<DimensionedTensorType>()) {
+      return cast->scalarType();
+    } else if (auto cast = type->cast<FloatType>()) {
+      return c10::ScalarType::Float;
+    } else if (auto cast = type->cast<IntType>()) {
+      return c10::ScalarType::Long;
+    } else if (auto cast = type->cast<BoolType>()) {
+      return c10::ScalarType::Bool;
+    }
+    return c10::nullopt;
+  }
+
+  c10::ScalarType getScalarType(Node *node) {
+    c10::ScalarType ret = c10::ScalarType::Undefined;
+    auto& args = node->schema().arguments();
+    for (size_t i = 0 ; i < args.size() ; i++ ) { //auto arg : args ) {
+      auto arg = args[i];
+      auto inputDtype = typeToScalarType(node->inputs()[i]->type());
+      if (inputDtype) {
+        if (ret == c10::ScalarType::Undefined) {
+          ret = inputDtype.value();
+        } else {
+          ret = c10::promoteTypes(ret, inputDtype.value());
```

Comment:
we usually just write `*inputDtype` fwiw

Reply:
fixed